src/api/spec/factories/project.rb

```diff
@@ -11,18 +11,13 @@
 
     factory :project_with_package do
       transient do
-        package_name nil
+        sequence(:package_name) { |n| "package_#{n}" }
         create_patchinfo false
         maintainer nil
       end
 
       after(:create) do |project, evaluator|
-        new_package = if evaluator.package_name
-                        create(:package, project_id: project.id, name: evaluator.package_name)
-                      else
-                        create(:package, project_id: project.id)
-                      end
-        project.packages << new_package
+        project.packages << create(:package, project_id: project.id, name: evaluator.package_name)
```

Comment:
But you are introducing a dependency between this factory and package factory...I mean, if someone changes the default way of package name is calculated, then you are overwriting it without knowing it and being maybe not compatible, isn' it?